kentikapi/users_integration_test.go

```diff
@@ -194,6 +195,108 @@ func TestClient_GetAllUsers(t *testing.T) {
 	}
 }
 
+func TestClient_GetUser_WithTimeout(t *testing.T) {
+	tests := []struct {
+		name            string
+		responses       []testutil.HTTPResponse
+		expectedResult  *models.User
+		expectedError   bool
+		expectedTimeout bool
+		// timeout and expected requests are strictly connected. The server sleeps for 1 ms before each response.
+		timeout      *time.Duration
+		expectedReqs int
+	}{
+		{
+			name: "default timeout",
+			responses: []testutil.HTTPResponse{
+				testutil.NewErrorHTTPResponse(http.StatusBadGateway),
+				testutil.NewErrorHTTPResponse(http.StatusBadGateway),
+				{StatusCode: http.StatusBadRequest, Body: `{"error":"Bad Request"}`},
+			},
+			expectedError:   true,
+			expectedTimeout: false,
+			expectedReqs:    3,
+		}, {
+			name: "status bad request",
```

Comment:
I don't see additional value in this test case when comparing to _TestClient_GetUser.status bad request_. Please remove that test case.

Reply:
Done